pkg/cmd/dev/util.go

```diff
@@ -105,11 +107,74 @@ func addCommonTestFlags(cmd *cobra.Command) {
 	cmd.Flags().Duration(timeoutFlag, 0*time.Minute, "timeout for test")
 }
 
-func ensureBinaryInPath(bin string) error {
+func ensureBinaryInPath(dev *dev, bin string) error {
 	if !isTesting {
-		if _, err := exec.LookPath(bin); err != nil {
+		if _, err := dev.exec.LookPath(bin); err != nil {
 			return errors.Newf("Could not find %s in PATH", bin)
 		}
 	}
 	return nil
 }
+
+// setupPath removes the ccache directory from PATH to prevent it writing files
+// outside of the bazel sandbox. This function is called only once to prevent
+// multiple unnecessary calls.
+func setupPath(dev *dev) error {
+	var once sync.Once
+	var err error
+	once.Do(func() {
+		err = setupPathReal(dev)
+	})
+	return err
+}
+
+// setupPathReal uses a list of known compiler names to check if they are
+// symlinks to `ccache`.
+func setupPathReal(dev *dev) error {
+	knownCompilers := []string{"cc", "gcc", "c++", "g++", "clang", "clang++"}
+	// datadriven uses Fprintln() to construct commands, adding '\n' to the end.
```

Comment:
Pretty unfortunate that this is necessary. Thanks for adding the comment to clarify though.